Source/WebGPU/WGSL/GlobalVariableRewriter.cpp

```diff
@@ -1554,7 +1554,7 @@ void RewriteGlobalVariables::insertParameters(AST::Function& function, const Vec
     };
     for (auto group : groups)
         insertParameter(group, argumentBufferParameterName(group));
-    if (!m_globalsUsingDynamicOffset.isEmpty()) {
+    if (!m_globalsUsingDynamicOffset.isEmpty() || (m_stage == ShaderStage::Fragment && m_callGraph.ast().usesFragDepth())) {
```

Comment:
this emits `__DynamicOffsets` buffer even when dynamic offsets are not used. It is because we pack the min / max depth into `__DynamicOffsets`